ide/services/ide-editor/src/main/java/org/eclipse/dirigible/runtime/ide/editor/services/EditorRestService.java

```diff
@@ -0,0 +1,154 @@
+
+    @Context
+    private HttpServletResponse response;
+
+    /**
+     * Finds all prettier config file paths inside a project
+     *
+     * @param workspaceName the workspace
+     * @param projectName   the project to search the config file in
+     * @param filePath      the config file name with extension
+     * @return the found prettier config paths
+     */
+    @GET
+    @Path("/prettier/config/{workspaceName}/{projectName}/{path:.*}")
+    public Response findPrettierConfig(
+            @PathParam("workspaceName") String workspaceName,
+            @PathParam("projectName") String projectName,
+            @PathParam("path") String filePath,
+            @Context HttpServletRequest request) {
+        String user = UserFacade.getName();
+        if (user == null) {
+            return createErrorResponseForbidden(NO_LOGGED_IN_USER);
+        }
+
+        String configFileName = ".prettierrc.json";
+        String configContentType = "application/json";
+        String separator = IRepository.SEPARATOR;
+
+        String projectPath = joinPathParts(separator, "users", user, workspaceName, projectName);
+        String queryPath = joinPathParts(separator, "prettier", "config", workspaceName, projectName, filePath);
+
+        RepositoryPath sourceFilePath = new RepositoryPath(projectPath + IRepository.SEPARATOR + filePath);
+        RepositoryPath sourceFolder = sourceFilePath.getParentPath();
+        String sourceFileName = sourceFilePath.getLastSegment();
+        ICollection lookupDirectory = repository.getCollection(sourceFolder.getPath());
+
+        if (!isPathNormalized(queryPath) || !isResourceInDirectory(lookupDirectory, sourceFileName)) {
+            return createErrorResponseNotFound(
+                    logMessage("There is resource at the specified path", queryPath)
+            );
+        }
+
+        while (isSubPath(projectPath, lookupDirectory.getPath())) {
+            String content = getConfigFileContent(lookupDirectory, configFileName);
+            if(content != null && !content.isEmpty()) {
+                return Response.ok(content).type(configContentType).build();
+            }
+
+            lookupDirectory = lookupDirectory.getParent();
+        }
+
+        return createErrorResponseNotFound(
+                logMessage("There is no config file for resource", queryPath)
+        );
+    }
+
+    private String logMessage(String reason, String queryPath) {
+        final String message = String.format("%s: %s", reason, queryPath);
+        logger.error(message);
+        return message;
+    }
+
+    private boolean isPathNormalized(String queryPath) {
+        return Objects.equals(FilenameUtils.normalize(queryPath), queryPath);
+    }
+
+    private boolean isResourceInDirectory(ICollection directory, String resourceName) {
+        return directory.getResource(resourceName).exists();
+    }
+
+    private String joinPathParts(String delimiter, String... args) {
```

Comment:
Couldn't such path methods use the Java `Path` interface or the Dirigible `RepositoryPath` interface?